src/views/CardStack/CardStackStyleInterpolator.js

```diff
@@ -20,6 +23,26 @@ import type { NavigationSceneRendererProps } from '../../TypeDefinition';
  */
 
 /**
+ * Format the sequence of the scenes
+ * @param scenes
+ * @param index
+ * @returns {NavigationScene}
+ */
+function formatScenes(scenes: Array<NavigationScene>, index: number) {
+  let scene = scenes[index];
+  if (!scene.isActive && index === scene.index) {
+    scenes = [].concat(scenes);
+    const activeIndex = scenes.findIndex(
+      (item: NavigationScene) => item.isActive
+    );
+    const activeScene = scenes[activeIndex];
+    scenes.splice(activeIndex, 1);
```

Comment:
Can this be refactored to use non-mutative `Array.prototype` methods?

Reply:
Before that I had created a new array。 my initial intention was to modify the queue order of the new array ， so I used `splice` here without any clash. Why must we use non-mutative medthod.